datumaro/components/merge/exact_merge.py

```diff
@@ -126,19 +126,22 @@ def _merge_media(
             not item_b.media or isinstance(item_b.media, MediaElement)
         ):
             if isinstance(item_a.media, MediaElement) and isinstance(item_b.media, MediaElement):
-                if (
-                    item_a.media.path
-                    and item_b.media.path
-                    and item_a.media.path != item_b.media.path
-                ):
+                item_a_path = getattr(item_a.media, "path", None)
+                item_b_path = getattr(item_b.media, "path", None)
+
+                if item_a_path and item_b_path and item_a_path != item_b_path:
                     raise MismatchingMediaPathError(
-                        (item_a.id, item_a.subset), item_a.media.path, item_b.media.path
+                        (item_a.id, item_a.subset), item_a_path, item_b_path
                     )
 
-                if item_a.media.path:
+                if item_a_path:
                     media = item_a.media
-                else:
+                elif item_b_path:
                     media = item_b.media
+                else:
+                    raise MismatchingMediaError(
+                        (item_a.id, item_a.subset), item_a.media, item_b.media
+                    )
```

Comment:
For the better readability, I suggest
```suggestion
                # Move this to L136
                elif item_a_path is None and item_b_path is None:
                    raise MismatchingMediaError(
                        (item_a.id, item_a.subset), item_a.media, item_b.media
                    )
                 media = item_a.media if item_a_path else item_b.media
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/944/commits/e8a0e80006a87d811ee6ca8a9ca542450de5ca5f